src/pymor/analyticalproblems/expressions.py

```diff
@@ -424,6 +477,11 @@ def numpy_expr(self):
         index = ['...'] + [repr(i) for i in self.index]
         return f'{self.base.numpy_expr()}[{",".join(index)}]'
 
+    def fenics_expr(self, params):
+        if len(self.base.shape) != 1:
+            raise NotImplementedError
+        return np.vectorize(lambda x: x.fenics_expr(params)[self.index[0]])(self.base)
```

Comment:
I would suspect that by now

```suggestion
        return self.base.fenics_expr(params)[self.index]
```

should work?